pkg/ibmvpc/parser.go

```diff
@@ -634,12 +634,28 @@ func getNACLconfig(rc *datamodel.ResourcesContainerModel,
 }
 
 func getTgwObjects(c *datamodel.ResourcesContainerModel,
-	res map[string]*vpcmodel.VPCConfig) map[string]*TransitGateway {
+	res map[string]*vpcmodel.VPCConfig, resourceGroup string) map[string]*TransitGateway {
 	tgwMap := map[string]*TransitGateway{} // collect all tgw resources
 	for _, tgwConn := range c.TransitConnectionList {
 		tgwUID := *tgwConn.TransitGateway.Crn
 		tgwName := *tgwConn.TransitGateway.Name
 		vpcUID := *tgwConn.NetworkID
+
+		// filtering by resourceGroup
+		if resourceGroup != "" {
+			shouldSkip := true
```

Comment:
This will filter TGW also if the resource group is unknown (the TGW object missing from the config object), and not only if it does not match.
In the test you added "multi_resource_groups" this is the case.. the config object does not contain any TGW objects, only Transit Connection objects..
I would consider at least a warning for filtering due to unknown RG , rather than actual "no match" filtering.